File: 2/2_cuatrimestre/PDOO/practicas/practica 4/deepSpace_4/src/deepspace/PowerEfficientSpaceStation.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 *
 * @author carlos
 */
public class PowerEfficientSpaceStation extends SpaceStation{
    
    private final float EFFICIENCYFACTOR;
    
    public PowerEfficientSpaceStation(SpaceStation station){
        super(station);
        EFFICIENCYFACTOR = 1.10F;
    }
    
    public float fire(){
        return super.fire()*EFFICIENCYFACTOR;
    }
    
    public float protection(){
        return super.protection()*EFFICIENCYFACTOR;
    }
    
    public Transformation setLoot(Loot loot){
        super.setLoot(loot);
        return Transformation.NOTRANSFORM;
        
    }
    
    public PowerEfficientSpaceStationToUI getUIversion(){
            return new PowerEfficientSpaceStationToUI(this);
    }
    
    @Override
    public String toString(){
        return getUIversion().toString();
    }
}

File: 2/2_cuatrimestre/PDOO/practicas/practica 4/deepSpaceRuby/PowerEfficientSpaceStation.rb
require_relative 'Transformation'
require_relative 'SpaceStation'
require_relative 'PowerEfficientSpaceStationToUI'

module Deepspace
    class PowerEfficientSpaceStation < SpaceStation
        def initialize (station)
            @EFFICIENCYFACTOR = 1.10
            newCopy(station)
        end


        def fire
            super * @EFFICIENCYFACTOR
        end

        def protection
            super * @EFFICIENCYFACTOR
        end
        
        def setLoot(loot)
            super
            Transformation::NOTRANSFORM
        end

        
        def getUIversion
            PowerEfficientSpaceStationToUI.new(self)
        end

        def to_s
            return getUIversion.to_s
        end
    end
end